dask/array/core.py

```diff
@@ -4480,8 +4480,7 @@ def handle_out(out, result):
             " Received type %s, expected Dask Array" % type(out).__name__
         )
         raise NotImplementedError(msg)
-    else:
-        return result
+    return result
```

Comment:
While this should eventually yields the correct result after a Dask computation, I think it follows NumPy behavior to return the actual `out` object under the `if isinstance(out, Array):` branch, that way the returned object is the same as the object passed to the `out=` argument.